src/go/k8s/api/redpanda/v1alpha1/redpanda_conversion.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+package v1alpha1
+
+import (
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	v1alpha2 "github.com/redpanda-data/redpanda-operator/src/go/k8s/api/redpanda/v1alpha2"
+)
+
+// SetupWebhookWithManager will setup the manager to manage the webhooks
+func (r *Redpanda) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// ConvertTo converts Redpanda to the Hub version (v1alpha2).
+func (src *Redpanda) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha2.Redpanda)
+
+	dst.ObjectMeta = src.ObjectMeta
+
+	dst.Status = src.Status
+
+	dst.Spec.Migration = src.Spec.Migration
+	dst.Spec.ChartRef = src.Spec.ChartRef
+
+	{
```

Comment:
I'm a bit confused about why we're creating a new scope here?

I think you could remove a lot of this code and instead marshal `src` to JSON and the unmarshal it into `dst`? You'll need to remove the incorrectly typed boolean before and after but everything else should "just work".